inspec/a2-iam-v2-upgrade-legacy-v1-policies/controls/base.rb

```diff
@@ -0,0 +1,322 @@
+    "viewer",
+    "ingest",
+    "project-owner"
+]
+
+include_controls 'a2-api-integration' do
+  # stuff that changed with the changes in default profiles
+  # see authz-access-control-1 for the IAM v2 behaviour
+  skip_control 'authz-access-control-iam-v1'
+
+  # this is using the v1 policy API, which will be disabled when using v2
+  skip_control 'authz-api-crud-1'
+end
+
+# these must be run after a2-api-integration because those
+# tests assume that the legacy policies exist
+include_controls 'remove-legacy-policies'
+
+title 'IAM v2.1 migration from v1 API integration tests'
+
+control 'upgrade-iam-v2-1' do
+  title 'IAM v2 Policies'
+  desc 'test IAM v2 policy API post migration'
+
+  describe 'list policies' do
+    it 'includes the legacy policies we expect' do
+      resp = automate_api_request('/apis/iam/v2beta/policies')
+      expect(resp.http_status).to eq 200
+
+      all_policies = resp.parsed_response_body[:policies]
+      legacy_policies = all_policies.select{ |p| /^\[Legacy\]/.match(p[:name]) }
+      expect(legacy_policies.length).to eq 9
+    end
+  end
+end
+
+control 'upgrade-iam-v2-2' do
+  title 'authz_access_control variants for IAM v2'
+  desc <<EOF
+In a2-api-integration, we've singled out a few checks whose results differ for
+IAM v2. In this control, they're included with their NEW results.
+EOF
+
+  NON_ADMIN_USERNAME_V2 = 'inspec_test_non_admin_with_iam_v2'
+
+  DEFAULT_PROJECT_ID = "default"
+
+  describe 'AuthZ access control' do
+    before(:all) do
+      create_non_admin_request = automate_api_request(
+        '/api/v0/auth/users',
+        http_method: 'POST',
+        request_body: {
+          'name': NON_ADMIN_USERNAME_V2,
+          'username': NON_ADMIN_USERNAME_V2,
+          'password': ENV['AUTOMATE_API_DEFAULT_PASSWORD'] || 'chefautomate',
+        }.to_json
+      )
+      expect(create_non_admin_request.http_status.to_s).to match(/200|409/)
+
+      test_token_request = automate_api_request(
+        '/api/v0/auth/tokens',
+        http_method: 'POST',
+        request_body: {
+          'description': 'inspec_test_token',
+          'active': true
+        }.to_json
+      )
+      TEST_TOKEN_V2 = test_token_request.parsed_response_body[:value]
+      TEST_TOKEN_ID_V2= test_token_request.parsed_response_body[:id]
+
+      expect(test_token_request.http_status).to eq(200)
+    end
+
+    after(:all) do
+      delete_non_admin_request = automate_api_request(
+        "/api/v0/auth/users/#{NON_ADMIN_USERNAME_V2}",
+        http_method: 'DELETE',
+      )
+      expect(delete_non_admin_request.http_status.to_s).to match(/200|404/)
+
+      delete_token_request = automate_api_request(
+        "/api/v0/auth/tokens/#{TEST_TOKEN_ID_V2}",
+        http_method: 'DELETE',
+      )
+      expect(delete_token_request.http_status.to_s).to match(/200|404/)
+    end
+
+    # compliance:profiles for client calls should be permitted by default
+    describe '/api/v0/compliance/profiles/search as client' do
```

Comment:
are these just testing that default legacy policies still work after migration?

Reply:
I think so